### PR TITLE
Feature/141 fix CVE new

### LINKED ISSUE #141: Fix security issues regarding RCE for Spring Core
The general recommendation from Apereo is to update to newest Version: 6.5.2

### PR DESCRIPTION
**Pull request by Subaro**
Resolves #141

### CODE REVIEW

File: app/src/main/java/de/triology/cas/services/EtcdClientFactory.java [hunk 1]
```diff
@@ -2,23 +2,19 @@
...
-@Component
 public class EtcdClientFactory {
 
     private static final String DEFAULT_NOTE_MASTER_FILE = "/etc/ces/node_master";
 
-    @Bean
-    public EtcdClient createEtcdClient() {
-        return createEtcdClient(DEFAULT_NOTE_MASTER_FILE);
+    public EtcdClient createDefaultFactory() {
+        return createEtcdClient(EtcdClientFactory.DEFAULT_NOTE_MASTER_FILE);
```
Comment (robertauer) [inline app/src/main/java/de/triology/cas/services/EtcdClientFactory.java:17]:
```suggestion
        return createEtcdClient(EtcdClientFactory.DEFAULT_NODE_MASTER_FILE);
```

File: app/src/main/java/de/triology/cas/services/EtcdClientFactory.java [hunk 2]
```diff
@@ -2,23 +2,19 @@
...
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileReader;
 import java.io.IOException;
 import java.net.URI;
 
-@Component
 public class EtcdClientFactory {
 
     private static final String DEFAULT_NOTE_MASTER_FILE = "/etc/ces/node_master";
```
Comment (robertauer) [inline app/src/main/java/de/triology/cas/services/EtcdClientFactory.java:14]:
```suggestion
    private static final String DEFAULT_NODE_MASTER_FILE = "/etc/ces/node_master";
```

File: app/src/main/java/de/triology/cas/services/EtcdClientFactory.java [hunk 3]
```diff
@@ -2,23 +2,19 @@
...
 
-    @Bean
-    public EtcdClient createEtcdClient() {
-        return createEtcdClient(DEFAULT_NOTE_MASTER_FILE);
+    public EtcdClient createDefaultFactory() {
```
Comment (robertauer) [inline app/src/main/java/de/triology/cas/services/EtcdClientFactory.java:16]: Why did the name change from `createEtcdClient()` to `createDefaultFactory()`? The function returns a client and not a factory, so the name should probably stay as it is, I think.

File: app/src/test/java/de/triology/cas/ldap/resolvers/MemberOfGroupResolverTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    @Mock
+    private Principal principal;
+
+    @Mock
+    private LdapEntry ldapEntry;
+
+    @Test
+    public void resolveGroupsWithMemberDn() {
+        LdapAttribute attribute = mock(LdapAttribute.class);
+        when(attribute.getStringValues()).thenReturn(Arrays.asList("cn=a", "cn=b,local", "c=c,c=c", "dc=d,o=d"));
```
Comment (robertauer) [inline app/src/test/java/de/triology/cas/ldap/resolvers/MemberOfGroupResolverTest.java:31]:
```suggestion
        when(attribute.getStringValues()).thenReturn(Arrays.asList("cn=b,local", "c=c,c=c", "cn=a", "dc=d,o=d"));
```
Maybe change order so it is not already ordered

**Review by robertauer — CHANGES_REQUESTED** [review]

File: app/etc/cas/config/log4j2.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
 <Configuration monitorInterval="5" packages="org.apereo.cas.logging,de.triology.cas.logging">
     <Properties>
         <Property name="baseDir">logs</Property>
-        <Property name="ces.log.level">error</Property>
+        <Property name="ces.log.level">trace</Property>
```
[the remark this answers is not in the document]
Comment (Subaro) [reply app/etc/cas/config/log4j2.xml:6]: This only affects the local development when running the CAS via `:/gradlew run`. A log level of `trace` is preferred when debugging

File: app/gradle.properties [hunk 6]
```diff
@@ -1,20 +1,26 @@
-# Versions
-cas.version=6.3.7.4
-springBootVersion=2.4.1
+# CAS server version
+cas.version=6.5.2
+springBootVersion=2.6.6
+
+# Control the version of the embedded Apache Tomcat server
+tomcatVersion=9.0.41
```
[the remark this answers is not in the document]
Comment (Subaro) [reply app/gradle.properties:6]: Yeah you are right. I simply forgot that.

[the remark this answers is not in the document]
Comment (Subaro) [reply app/gradle.properties:6]: Done

File: app/src/main/java/de/triology/cas/services/EtcdClientFactory.java (hunk 2 above)
Comment (Subaro) [reply app/src/main/java/de/triology/cas/services/EtcdClientFactory.java:14]: Done

File: app/src/main/java/de/triology/cas/services/EtcdClientFactory.java (hunk 3 above)
Comment (Subaro) [reply app/src/main/java/de/triology/cas/services/EtcdClientFactory.java:16]: Changed to `createDefaultClient`. Done

File: app/src/test/java/de/triology/cas/ldap/resolvers/MemberOfGroupResolverTest.java (hunk 4 above)
Comment (Subaro) [reply app/src/test/java/de/triology/cas/ldap/resolvers/MemberOfGroupResolverTest.java:31]: The order of the elements it completely indifferent.

**Review by robertauer — APPROVED** [review]

**Pull request merged**